homeassistant/components/tuya/__init__.py

```diff
@@ -107,12 +144,24 @@ async def async_setup_entry(hass, entry):
 
     hass.data[DOMAIN] = {
         TUYA_DATA: tuya,
+        TUYA_DEVICES_CONF: entry.options.copy(),
         TUYA_TRACKER: None,
         ENTRY_IS_SETUP: set(),
         "entities": {},
         "pending": {},
+        entry.entry_id: {
```

Comment:
Since we only allow one entry, its ok to store "listener" without the `entry.entry_id`